android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingInsightScreen.kt

```diff
@@ -16,80 +16,171 @@
-  unsyncedResources: List<Pair<String, Int>>,
-  onDismissRequest: () -> Unit,
+  fullName: String?,
+  team: String?,
+  locality: String?,
+  userName: String?,
+  organization: String?,
+  careTeam: String?,
+  location: String?,
+  appVersionCode: String,
+  appVersion: String,
+  buildDate: String,
+  unsyncedResourcesFlow: MutableSharedFlow<List<Pair<String, Int>>>,
+  navController: NavController,
+  onRefreshRequest: () -> Unit,
 ) {
-  Box(Modifier.clip(RectangleShape).fillMaxWidth().background(Color.White)) {
-    Dialog(onDismissRequest = onDismissRequest) {
-      Column(
-        modifier = Modifier.fillMaxWidth().wrapContentHeight().background(Color.White),
-        horizontalAlignment = Alignment.CenterHorizontally,
-      ) {
+  val unsyncedResources = unsyncedResourcesFlow.collectAsState(initial = listOf()).value
+
+  Scaffold(
+    topBar = {
+      TopAppBar(
+        title = { Text(text = stringResource(R.string.insights)) },
+        navigationIcon = {
+          IconButton(onClick = { navController.popBackStack() }) {
+            Icon(
+              imageVector = Icons.Filled.ArrowBack,
+              modifier = Modifier.testTag(USER_INSIGHT_TOP_APP_BAR),
+              contentDescription = null,
+            )
+          }
+        },
+        contentColor = Color.White,
+        backgroundColor = MaterialTheme.colors.primary,
+      )
+    },
+    backgroundColor = Color.White,
+  ) { paddingValues ->
+    LazyColumn(
+      Modifier.fillMaxSize().padding(paddingValues).background(Color.White),
+      horizontalAlignment = Alignment.Start,
+      contentPadding = PaddingValues(16.dp),
+    ) {
+      item {
         Text(
           text = stringResource(id = R.string.unsynced_resources),
-          modifier = Modifier.padding(11.dp),
-          style = TextStyle(color = Color.Black, fontSize = 20.sp),
-          fontWeight = FontWeight.Light,
+          style = TextStyle(color = Color.Black, fontSize = 18.sp),
+          fontWeight = FontWeight.Bold,
         )
-        LazyColumn(modifier = Modifier.wrapContentHeight()) {
-          items(unsyncedResources) { language ->
-            Box(Modifier.fillMaxWidth().padding(15.dp)) {
-              Text(
-                text = language.first,
-                modifier = Modifier.align(Alignment.CenterStart),
-                fontWeight = FontWeight.Light,
-              )
-              Text(
-                text = language.second.toString(),
-                modifier = Modifier.align(Alignment.CenterEnd),
-              )
-            }
-            Spacer(modifier = Modifier.padding(1.dp))
-          }
+        if (unsyncedResources.isEmpty()) {
+          Divider(thickness = 1.dp, color = LoginDarkColor.copy(alpha = 0.67f))
         }
-        Column(Modifier.wrapContentWidth().wrapContentHeight().padding(4.dp)) {
+      }
+      if (unsyncedResources.isNotEmpty()) {
+        items(unsyncedResources) { unsynced ->
+          UnsyncedDataView(
```

Comment:
The UI for  `UnsyncedDataView`,  `AppInfoView`, `AssignmentInfoView`,  `UserInforView`, and `DeviceInfoView` can all share the same UI component. The UI for each section of the Insight screen looks similar. Implement a single Composable function to be re-used in all sections with just the content updated to improve code maintainability.